extensions/spring-xd-extension-batch/src/main/java/org/springframework/batch/step/tasklet/x/AbstractProcessBuilderTasklet.java

```diff
@@ -102,34 +163,37 @@ public RepeatStatus execute(StepContribution contribution, ChunkContext chunkCon
 			logger.error(msg);
 		}
 		finally {
-			printLog(commandName, commandLog, exitCode);
-			StringBuilder firstException = new StringBuilder();
-			if (exitCode != 0) {
-				for (String line : commandLog) {
-					if (firstException.length() == 0) {
-						if (line.contains("Exception")) {
-							firstException.append(line).append("\n");
-						}
-					}
-					else {
-						if (line.startsWith("\t")) {
-							firstException.append(line).append("\n");
-						}
-						else {
-							break;
-						}
-					}
+			if (p != null) {
+				p.destroy();
+			}
+			commandOut = getProcessOutput(out);
+			commandErr = getProcessOutput(err);
+			printLog(commandName, commandOut, commandErr);
```

Comment:
Is it too crazy to store all the lines in the step execution?  It is the one central place that we do have for storing state.